src/scalar.rs

```diff
@@ -62,7 +62,7 @@ impl<'a> Scalar<'a> {
     }
 
     /// View the raw data
-    pub fn view_data(&self) -> &'a [u8] {
+    pub fn view_data(self) -> &'a [u8] {
```

Comment:
Drive by, but you probably want to make many methods here with `#[inline]` to enable cross-crate inlining. 

https://nnethercote.github.io/perf-book/inlining.html

> If you are not using link-time optimization, functions will never be inlined across crates.

note that even `release` profile does not enable lto.

Reply:
Thanks for the suggestion, it I applied it: https://github.com/rakaly/jomini/commit/af787789de508c743369843748c2a317422522e7